lib/include/TransportDescriptor.hpp

```diff
@@ -0,0 +1,2033 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @file
+ */
+
+#ifndef _FAST_DDS_QOS_PROFILES_MANAGER_TRANSPORT_DESCRIPTOR_HPP_
+#define _FAST_DDS_QOS_PROFILES_MANAGER_TRANSPORT_DESCRIPTOR_HPP_
+
+#include <string>
+#include <vector>
+
+namespace eprosima {
+namespace qosprof {
+namespace transport_descriptor {
+
+/************************************************************************/
+/* Print functions                                                      */
+/************************************************************************/
+
+/**
+ * @brief Parse XML file and print every transport descriptor profile.
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ *
+ * @return std::string XML section containing the transport descriptor profiles.
+ *
+ * @throw file_not_found Exception if the provided XML file is not found/readable.
+ */
+std::string print(
+        const std::string& xml_file);
+
+/**
+ * @brief Parse XML file and print specific transport descriptor profile.
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] transport_descriptor_id Transport descriptor profile identifier.
+ *
+ * @return std::string XML section containing the specific transport descriptor profile.
+ *
+ * @throw file_not_found Exception if the provided XML file is not found/readable.
+ * @throw element_not_found Exception if the specified transport descriptor profile is not found in the XML file.
+ */
+std::string print(
+        const std::string& xml_file,
+        const std::string& transport_descriptor_id);
+
+/**
+ * @brief Parse XML file and print the specific Transport Descriptor kind.
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] transport_descriptor_id Transport descriptor profile identifier.
+ *
+ * @return std::string Transport Descriptor kind
+ *
+ * @throw file_not_found Exception if the provided XML file is not found/readable.
+ * @throw element_not_found Exception if the specified transport descriptor profile is not found in the XML file.
+ */
+std::string print_kind(
+        const std::string& xml_file,
+        const std::string& transport_descriptor_id);
+
+/**
+ * @brief Parse XML file and print the specific Transport Descriptor send buffer size.
+ *
+ * @param[in] xml_file Absolute/relative path to the XML file.
+ * @param[in] transport_descriptor_id Transport descriptor profile identifier.
+ *
+ * @return uint32_t Transport descriptor send buffer size.
+ *
+ * @throw file_not_found Exception if the provided XML file is not found/readable.
+ * @throw element_not_found Exception if the specified transport descriptor profile is not found in the XML file or the
+ *        send buffer size element has not been set in the profile.
+ */
```

Comment:
I would have more than one exception, but I'd also keep the what as a help of sorts.

Reply:
I do not mind having more than one exception either. I think this will also be at least a partially reliant on what functionalities the underlying libraries provide for error handling.

Reply:
This decition would be taken based on the exceptions thrown by the final XML library used.

Last tests would be performed to determine the library.

Reply:
Finally it was determined that both libraries do not throw multiple exceptions neither the complete set of exceptions designed, so we should implement a Exception class to throw them.